sound/soc/sof/pm.c

```diff
@@ -332,10 +332,13 @@ int snd_sof_prepare(struct device *dev)
 	struct snd_sof_dev *sdev = dev_get_drvdata(dev);
 
 #if defined(CONFIG_ACPI)
-	sdev->s0_suspend = acpi_target_system_state() == ACPI_STATE_S0;
+	if (acpi_target_system_state() == ACPI_STATE_S0)
+		sdev->system_suspend_target = SOF_SUSPEND_S0;
```

Comment:
we should probably call this SOF_SUSPEND_SOix.
There is no real reason to stick with ACPI conventions which make no sense really.